src/Elastic.Apm/Logging/ScopedLogger.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Concurrent;
+
+namespace Elastic.Apm.Logging
+{
+	internal class ScopedLogger : IApmLogger
+	{
+		public ScopedLogger(IApmLogger logger, string scope) => (Logger, Scope) = (logger, scope);
+
+		private ConcurrentDictionary<string, LogValuesFormatter> Formatters { get; } = new ConcurrentDictionary<string, LogValuesFormatter>();
+
+		public IApmLogger Logger { get; }
+
+		public string Scope { get; }
```

Comment:
I wonder if we can make Scope something more flexible so that we can scope not only by class name but the method name, transaction name, etc. Maybe we can make Scope in ScopedLogger to be a Map<string, string> and ScopedLogger 
 constructor will become ScopedLogger(IApmLogger logger, string scopedKey, string scopedValue) and a new contructor ScopedLogger(IApmLogger logger, Map<string, string> scopedKeyValues) to scope multiple (key. value) in one go. What do you think?

Reply:
I personally don't feel we need to be that granular with scoping. Typically the class name should be sufficient. We can extend the extension methods to include Caller Information for all messages (not just exceptions) instead?

Reply:
Agreed - we can wait and see what other needs arise.